websites/B/Board Game Arena/util.ts

```diff
@@ -79,8 +79,10 @@ export async function getMetadata<T>(
 	if (now - getCacheTime(key) > 1000) {
 		setCacheTime(key, now);
 		const data = await getPageVariable<T>(`gameui.${key}`, presence, isString);
-		setCachedItem(key, data);
-		return data;
+		if (data) {
+			setCachedItem(key, data);
+			return data;
+		} else return "" as T;
```

Comment:
Ensures that the result is `undefined` if the data doesn't exist the first time, rather than having one frame where the result is a string and then in the next, `undefined`
```suggestion
		if (data) {
			setCachedItem(key, data);
		}
		return data;
```